freeipa/src/test/java/com/sequenceiq/freeipa/flow/freeipa/backup/full/CreateFullBackupHandlerTest.java

```diff
@@ -0,0 +1,138 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.sequenceiq.cloudbreak.orchestrator.exception.CloudbreakOrchestratorFailedException;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.host.OrchestratorStateParams;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.Node;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.freeipa.backup.full.event.CreateFullBackupEvent;
+import com.sequenceiq.freeipa.flow.stack.StackEvent;
+import com.sequenceiq.freeipa.flow.stack.StackFailureEvent;
+import com.sequenceiq.freeipa.service.GatewayConfigService;
+import com.sequenceiq.freeipa.service.freeipa.flow.FreeIpaNodeUtilService;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+import reactor.bus.Event;
+
+@ExtendWith(MockitoExtension.class)
+class CreateFullBackupHandlerTest {
+
+    @Mock
+    private HostOrchestrator orchestrator;
+
+    @Mock
+    private GatewayConfigService gatewayConfigService;
+
+    @Mock
+    private StackService stackService;
+
+    @Mock
+    private FreeIpaNodeUtilService nodeService;
+
+    @InjectMocks
+    private CreateFullBackupHandler underTest;
+
+    @Test
+    public void testSelector() {
+        assertEquals(EventSelectorUtil.selector(CreateFullBackupEvent.class), underTest.selector());
+    }
+
+    @Test
+    public void testDefaultFailureEvent() {
+        Exception e = new Exception();
+
+        StackFailureEvent result = (StackFailureEvent) underTest.defaultFailureEvent(2L, e, new Event<>(new CreateFullBackupEvent(3L)));
+
+        assertEquals(2L, result.getResourceId());
+        assertEquals(e, result.getException());
+        assertEquals(FullBackupEvent.FULL_BACKUP_FAILED_EVENT.event(), result.selector());
+    }
+
+    @Test
+    public void testBackupSuccessful() throws CloudbreakOrchestratorFailedException {
+        Stack stack = mock(Stack.class);
+        when(stackService.getByIdWithListsInTransaction(2L)).thenReturn(stack);
+        Set<InstanceMetaData> metaDataSet = Set.of();
+        when(stack.getNotDeletedInstanceMetaDataSet()).thenReturn(metaDataSet);
+        Node node1 = mock(Node.class);
+        when(node1.getHostname()).thenReturn("node1");
+        Node node2 = mock(Node.class);
+        when(node2.getHostname()).thenReturn("node2");
+        Set<Node> nodes = Set.of(node1, node2);
```

Comment:
I might extract a creator method for that (another test also creates nodes), taking a set of names.

Reply:
you care about tests too much :)